src/Shared/Dictionary/AdaptiveCapacityDictionary.cs

```diff
@@ -95,6 +94,7 @@ internal AdaptiveCapacityDictionary(IEnumerable<KeyValuePair<TKey, TValue>> valu
         {
             _comparer = comparer ?? EqualityComparer<TKey>.Default;
 
+            Debug.Assert(capacity <= DefaultArrayThreshold);
```

Comment:
Why doesn't this ctor "fallback" to dictionary with a big enough size?

Reply:
This constructor is only used in tests.

Reply:
It should just call the other ctor, no reason to have split code for testing and non-testing

Reply:
Let's see if I can just delete it, the test usage doesn't look novel.